util/metrics/otel.py

```
@@ -47,14 +171,19 @@ def decorate(func):
+                ctx = get_tracecontext(request.headers)
                 tracer = trace.get_tracer(__name__)
                 name = span_name if span_name else func.__name__
-                with tracer.start_as_current_span(name) as span:
+                with tracer.start_as_current_span(name, context=ctx) as span:
                     try:
+                        span.set_status(StatusCode.OK)
+                        span.set_attribute("function", name)
+                        span.set_attribute("args", str(args))
+                        span.set_attribute("kwargs", str(kwargs))
                         return func(*args, **kwargs)
                     except Exception as e:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)